stdlib/strtoul.c

```diff
@@ -5,58 +5,99 @@
  *
  * strtoul, strtol
  *
- * Copyright 2017-2018 Phoenix Systems
- * Author: Jakub Sejdak, Pawel Pisarczyk, Michal Miroslaw
+ * Copyright 2017-2018, 2023 Phoenix Systems
+ * Author: Jakub Sejdak, Pawel Pisarczyk, Michal Miroslaw, Gerard Swiderski
  *
  * This file is part of Phoenix-RTOS.
  *
  * %LICENSE%
  */
 
+#include <errno.h>
 #include <stdlib.h>
 #include <ctype.h>
+#include <limits.h>
 
 
 unsigned long int strtoul(const char *nptr, char **endptr, int base)
 {
-	unsigned long int t, v = 0;
+	unsigned long int cutoff, t, result = 0;
+	int cutlim, width = 0, negative = 0;
```

Comment:
`cutlim` should probably be like `cutoff` of type `unsigned long int`

Reply:
The remainder of division`cutlim` is in range of `2 ≤ base ≤ 36`. The right side of assignment is `unsigned long int`, after assignment it is truncated to `int` and because of the range 2 .. 36, truncation does not occur. The result of modulo is `0 ≤ cutlim ≤ 35` (depending on `base`), for me`int` type here is ok, but ... The other thing is variable `t` which shall be `int` and not `unsigned long int`, depending on single digit range `'0' ≤ t ≤ 'z'`, the check `t > cutlim` is then valid.